Realm/RLMSyncSession.h

```diff
@@ -30,11 +30,57 @@ typedef NS_ENUM(NSUInteger, RLMSyncSessionState) {
     RLMSyncSessionStateInvalid
 };
 
+/**
+ The type of data transfer that a particular sync session notifier block will track.
+ */
+typedef NS_ENUM(NSUInteger, RLMSyncNotifierDirection) {
+    /// For monitoring upload progress.
+    RLMSyncNotifierDirectionUpload,
+    /// For monitoring download progress.
+    RLMSyncNotifierDirectionDownload,
+};
+
+/**
+ The desired behavior of a particular sync session notifier block.
+ */
+typedef NS_ENUM(NSUInteger, RLMSyncNotifierMode) {
+    /// The block will be called forever, or until the user unregisters it.
+    /// It will always report the latest number of transferred bytes, and the
+    /// latest number of total bytes to be transferred.
+    RLMSyncNotifierModeAlwaysReportLatest,
+    /// The block will, upon registration, store the total number of bytes
+    /// to be transferred. When invoked, it will always report the latest number
+    /// of transferred bytes out of that original number of bytes to be
+    /// transferred. When the number of transferred bytes reaches or exceeds the
+    /// number of bytes to be transferred, the block will be unregistered.
+    RLMSyncNotifierModeProgressIndicator,
```

Comment:
The naming of these enum cases doesn't seem great. It's also different than the terminology used in the Swift API.